be/src/olap/rowset/segment_v2/page_handle.h

```diff
@@ -46,42 +46,49 @@ class PageHandle {
             : _is_data_owner(false), _cache_data(std::move(cache_data)) {}
 
     // Move constructor
-    PageHandle(PageHandle&& other) noexcept
-            : _is_data_owner(false),
-              _data(std::move(other._data)),
-              _cache_data(std::move(other._cache_data)) {
+    PageHandle(PageHandle&& other) noexcept : _cache_data(std::move(other._cache_data)) {
         // we can use std::exchange if we switch c++14 on
         std::swap(_is_data_owner, other._is_data_owner);
+        _data.swap(other._data);
     }
 
     PageHandle& operator=(PageHandle&& other) noexcept {
         std::swap(_is_data_owner, other._is_data_owner);
-        _data = std::move(other._data);
+        _data.swap(other._data);
         _cache_data = std::move(other._cache_data);
         return *this;
     }
 
     ~PageHandle() {
         if (_is_data_owner && _data.size > 0) {
-            delete[] _data.data;
             ExecEnv::GetInstance()->page_no_cache_mem_tracker()->consume(-_data.size);
+        } else {
+            DCHECK(_data.size == 0);
         }
     }
 
     // the return slice contains uncompressed page body, page footer, and footer size
-    Slice data() const {
+    char* data() const {
```

Comment:
why not return slice ?

Reply:
It seems more graceful that DataPage and Slick are independent?

![image](https://github.com/apache/doris/assets/13197424/bb006bb8-6e1a-4a4e-840a-ef4921077919)